x/zkism/keeper/events.go

```diff
@@ -4,42 +4,27 @@ import (
 	"encoding/hex"
 	"fmt"
 
-	errorsmod "cosmossdk.io/errors"
 	"github.com/celestiaorg/celestia-app/v6/x/zkism/types"
-	"github.com/celestiaorg/go-square/v2/share"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // EmitCreateISMEvent emits a typed event to signal creation of a new ism.
-func EmitCreateISMEvent(ctx sdk.Context, ism types.ZKExecutionISM) error {
-	namespace, err := share.NewNamespaceFromBytes(ism.Namespace)
-	if err != nil {
-		return errorsmod.Wrapf(types.ErrInvalidNamespace, "failed to parse namespace from bytes: %x", ism.Namespace)
-	}
-
-	return ctx.EventManager().EmitTypedEvent(&types.EventCreateZKExecutionISM{
+func EmitCreateISMEvent(ctx sdk.Context, ism types.InterchainSecurityModule) error {
+	return ctx.EventManager().EmitTypedEvent(&types.EventCreateInterchainSecurityModule{
 		Id:                  ism.Id,
 		Owner:               ism.Owner,
-		CelestiaHeaderHash:  encodeHex(ism.CelestiaHeaderHash),
-		CelestiaHeight:      ism.CelestiaHeight,
-		StateRoot:           encodeHex(ism.StateRoot),
-		Height:              ism.Height,
-		Namespace:           namespace.String(),
-		SequencerPublicKey:  encodeHex(ism.SequencerPublicKey),
+		State:               ism.State,
```

Comment:
when we update the field to be string again

```suggestion
		State:               encodeHex(ism.State),
```